Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -74,9 +74,11 @@ namespace AZ
         {
             if (m_pendingRecreation)
             {
+                VkSwapchainKHR oldSwapchain = m_nativeSwapChain;
                 ShutdownImages();
-                InvalidateNativeSwapChain();
                 CreateSwapchain();
+                // Destroy the old swapchain AFTER creating the new one (because it's used for building the new one)
+                InvalidateNativeSwapChain(oldSwapchain);
```

Comment:
You are destroying the oldSwapchain right away with these changes. Wouldn't this cause a crash? I had to fix issues around this problem by adding support to put the old swap chain on the release queue so it is destroyed after 3 frames. In order for this PR to go in I recommend we test Editor, Material Editor, game launcher while resizing them + minimizing them on PC and Vulkan. Jus to make sure we are not introducing a previously fixed crash.

Reply:
This seems to still be an issue, it does not happen a lot, but when resizing the window often enough, we still reliably get it to hang on a condition variable in `InvalidateNativeSwapChain`.
![image](https://github.com/o3de/o3de/assets/102576959/6b444e4b-993c-40ed-bb6c-5da1426a6afc)

Is that a know issue?